Source/DafnyCore/AST/ShouldCompileOrVerify.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Dafny; 
+
+public static class ShouldCompileOrVerify {
+
+  public static bool ShouldCompile(this ModuleDefinition module, Program program) {
+    if (program.UrisToCompile == null) {
+      program.UrisToCompile = ComputeUrisToCompile(program);
+    }
+
+    if (module.FullName == "_System") {
+      return true;
+    }
+
+    if (module is DefaultModuleDefinition) {
+      // If things from precompiled files live in the default module, that can cause downstream compilation issues:
+      // https://github.com/dafny-lang/dafny/issues/4009
```

Comment:
Is this meant to be a TODO/FIXME?

Reply:
The GH issue itself is meant to be the 'TODO', without putting an actual TODO in the code. I prefer this way of planning future work.